### PR TITLE
Implemented Finite State Machine

### PR DESCRIPTION
**Pull request by gregormaclaine**
This feature branch creates a FSM framework for the program that will make it much easier to have different views that can be switched between such as the menu and the game. This also involved clearing up the code for the menu and modularising some of the features in it like the button.

The most recent commits in main have been merged into it, meaning it includes the new additions of player health and the background.

This version is also fully functional on macOS amd64, so it needs testing on windows before mergining into the main branch.

### CODE REVIEW

**Review by egorvert — CHANGES_REQUESTED** [review]
Code is good and functional on Mac. It would be appreciated if you could add comments to your code to explain what it does so the rest of the team doesn't have to spend their precious time on figuring out what the hell you're doing.

Some bugs I noticed:
- Hitboxes are always shown and control / command B does not turn them off
- Resizing a window causes the character to fly off the screen

Tl;Dr
★★★★☆
Good feature rich commit but with little annotation and a few bugs. Also needs testing on Windows.

**Comment by gregormaclaine** [comment]
> Code is good and functional on Mac. It would be appreciated if you could add comments to your code to explain what it does so the rest of the team doesn't have to spend their precious time on figuring out what the hell you're doing.
> 
> Some bugs I noticed:
> 
> * Hitboxes are always shown and control / command B does not turn them off
> * Resizing a window causes the character to fly off the screen
> 
> Tl;Dr ★★★★☆ Good feature rich commit but with little annotation and a few bugs. Also needs testing on Windows.

**Comment by gregormaclaine** [comment]
Woops closed it by accident. Also I see the bug with resizing the window, but the hitbox outlines work perfectly find for me? They don't show up to begin with and toggle visibility with Ctrl-B.

**Comment by gregormaclaine** [comment]
All bugs and changes requested by Egor have been implemented.
Branch is now ready for merge
